src/content/getusermedia/record/js/main.js

```diff
@@ -32,7 +34,8 @@ recordButton.addEventListener('click', () => {
 
 const playButton = document.querySelector('button#play');
 playButton.addEventListener('click', () => {
-  const superBuffer = new Blob(recordedBlobs, {type: 'video/webm'});
+  const mimeType = codecPreferences.options[codecPreferences.selectedIndex].value.split(';', 1)[0];
```

Comment:
I would have preferred to use `mediaRecorder.mimeType` here but it is
```
"video/x-matroska;codecs=avc1,opus"
```
when one selects H264

Reply:
Also another note, IIRC the `ondataavailable` event listener's `event.data.type` should carry the actual `mimeType` used in the recording (which might be slightly different from the one specified):

https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/web_tests/fast/mediarecorder/MediaRecorder-requestData.html;l=68

Reply:
line 75 is for Safari which presumably only supports MP4 -- see https://webkit.org/blog/11353/mediarecorder-api/
It doesn't work in Chrome but that just means it will be filtered.

e.data.type is the same as mediaRecorder.mimeType